akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -197,4 +200,24 @@ object GraphStages {
     }
     override def toString: String = s"SingleSource($elem)"
   }
+
+  private[stream] final class FutureSource[T](val future: Future[T]) extends GraphStage[SourceShape[T]] {
+    ReactiveStreamsCompliance.requireNonNullElement(future)
+    val shape = SourceShape(Outlet[T]("future.out"))
+    val out = shape.out
```

Comment:
It more confirms that having `val out` is a nice thing to have though IMO :wink: